tests/framework/extensions/kubeapi/webhook/getwebhook.go

```diff
@@ -0,0 +1,34 @@
+package webhook
+
+import (
+	"context"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	clientV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	V1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GetWebhook is a helper function that uses the dynamic client to get a list of webhooks from a cluster
+func GetWebhook(client *rancher.Client, clusterID, resourceName string) (*V1.ValidatingWebhookConfiguration, error) {
+	dynamicClient, err := client.GetDownStreamClusterClient(clusterID)
+	if err != nil {
+		return nil, err
+	}
+	webHookResource := schema.GroupVersionResource{
+		Group:    "admissionregistration.k8s.io",
+		Version:  "v1",
+		Resource: "validatingwebhookconfigurations",
+	}
+
+	result, err := dynamicClient.Resource(webHookResource).Get(context.TODO(), resourceName, metav1.GetOptions{})
+
+	webhook := &V1.ValidatingWebhookConfiguration{}
+	err = clientV1.ConvertToK8sType(result.Object, webhook)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
So, the dynamic client's GET request returns an unstructured object. And it might be unified to convert this unstructured object to the Kubernetes object instead v1 object by using scheme convert. The other kubeapi packages should have a similar approach where you can take it as an example.

Reply:
Actually, now that I think about it more, could you check the above briefly? It might not be possible in this case.